### ISSUE TITLE
Process_getCommand() needs updation.

### ISSUE THREAD

**Issue by gnarendran** [htop-dev/htop#934]
1. In the original "Merged Command for Linux" commit, the flag `showMergedCommand` considered the flag `showThreadNames` to mean "display only the thread name (comm), but not exe/cmdline". In those days `showThreadNames` caused `cmdline` for Linux to only contain "comm". So the `Process_getCommand()` had been coded thus:

```c
const char* Process_getCommand(const Process* this) {
   if ((Process_isUserlandThread(this) && this->settings->showThreadNames) || !this->mergedCommand.str) {
      return this->cmdline;
   }
   return this->mergedCommand.str;
}
```

But it seems the current interpretation is to tie thread name exclusively to `showMergedCommand` and to display the entire "exe|comm|cmdline". This is also ok (as long as the user is aware that thread names would no longer be shown unless merge is chosen and has needed permission), except that `Process_getCommand()` should also be updated as follows:

```c
const char* Process_getCommand(const Process* this) {
   return this->mergedCommand.str ? this->mergedCommand.str : this->cmdline;
}
```

Otherwise `Process_getCommand()` returns raw cmdline for threads with `mergedCommand`. This affects searching/sorting/filterting and wrapping where `Process_getCommand` is used. This issue is readily seen by enabling both `showThreadNames` and `showMergedCommand`, disabling `hideUserlandThreads`, and searching for a thread name displayed - it fails to find it. Similarly, wrap on a merged thread tries to display raw `cmdline` and the results are not pretty.

2. In `htop.1.in` (man page) there are multiple descriptions of COMM and EXE - the latter ones could be removed, merged with the first ones.

Regards
Narendran Gopalakrishnan

**Comment by BenBE** [comment]
This may be related to #912 …

**Comment by BenBE** [comment]
@gnarendran Sorry for the delay with reviewing this issue. Please check if the current main branch still suffers from the issue you describe. As noted in #877/#912 limiting the display of the custom thread names to the merged view only was an accident that happened when the overall command line handling was unified across platforms.

If the first issue still persists, please feel free to open up a PR with your proposed fix, to make reviewing and testing easier.

The documentation issue should be fixed too.

**Comment by gnarendran** [comment]
@BenBE: The man page is fixed, thanks. As suggested I will open a new PR describing the current (after #877 / #912 ) behavior of "Command" column display and corresponding update needed in Process_getCommandStr().

Regards.
